src/components/learner-credit-management/tests/BudgetDetailPage.test.jsx

```diff
@@ -1158,4 +1200,138 @@ describe('<BudgetDetailPage />', () => {
       expect(remindButton).toBeDisabled();
     }
   });
+
+  it('cancels assignments in bulk', async () => {
+    EnterpriseAccessApiService.cancelContentAssignments.mockResolvedValueOnce({ status: 200 });
+    useParams.mockReturnValue({
+      budgetId: mockSubsidyAccessPolicyUUID,
+      activeTabKey: 'activity',
+    });
+    useBudgetRedemptions.mockReturnValue({
+      isLoading: false,
+      budgetRedemptions: mockEmptyBudgetRedemptions,
+      fetchBudgetRedemptions: jest.fn(),
+    });
+    useSubsidyAccessPolicy.mockReturnValue({
+      isInitialLoading: false,
+      data: mockAssignableSubsidyAccessPolicy,
+    });
+    useBudgetDetailActivityOverview.mockReturnValue({
+      isLoading: false,
+      data: {
+        contentAssignments: { count: 1 },
+        spentTransactions: { count: 0 },
+      },
+    });
+    useBudgetContentAssignments.mockReturnValue({
+      isLoading: false,
+      contentAssignments: {
+        count: 2,
+        results: [
+          {
+            uuid: 'test-uuid1',
+            contentKey: mockCourseKey,
+            contentQuantity: -19900,
+            learnerState: 'waiting',
+            recentAction: { actionType: 'assigned', timestamp: '2023-10-27' },
+            actions: [mockSuccessfulNotifiedAction],
+            errorReason: null,
+            state: 'allocated',
+          },
+          {
+            uuid: 'test-uuid2',
+            contentKey: mockCourseKey,
+            contentQuantity: -29900,
+            learnerState: 'waiting',
+            recentAction: { actionType: 'assigned', timestamp: '2023-11-27' },
+            actions: [mockSuccessfulNotifiedAction],
+            errorReason: null,
+            state: 'allocated',
+          },
+        ],
+        learnerStateCounts: [
+          { learnerState: 'waiting', count: 1 },
+          { learnerState: 'waiting', count: 1 },
+        ],
+        numPages: 1,
+        currentPage: 1,
+      },
+      fetchContentAssignments: jest.fn(),
+    });
+    renderWithRouter(<BudgetDetailPageWrapper />);
+    // const cancelRowAction = screen.getByTestId('datatable-select-column-checkbox-header');
+    const cancelRowAction = screen.getByTitle('Toggle All Current Page Rows Selected');
+    expect(cancelRowAction).toBeInTheDocument();
+    userEvent.click(cancelRowAction);
+    const cancelBulkActionButton = screen.getByText('Cancel (2)');
+    expect(cancelBulkActionButton).toBeInTheDocument();
+    userEvent.click(cancelBulkActionButton);
+    const modalDialog = screen.getByRole('dialog');
+    expect(modalDialog).toBeInTheDocument();
+    const cancelDialogButton = screen.getByRole('button', { name: 'Cancel assignments (2)' });
+    userEvent.click(cancelDialogButton);
+    expect(
+      EnterpriseAccessApiService.cancelContentAssignments,
+    ).toHaveBeenCalled();
+    await waitFor(
+      () => act(() => expect(screen.getByText('Assignments canceled (2)')).toBeInTheDocument()),
+    );
+  });
```

Comment:
This seems a bit funky to wrap an `act` with an `waitFor`. It seems like the test still passes with the `act`?